crates/starknet-devnet-types/src/rpc/transactions/broadcasted_declare_transaction_v2.rs

```diff
@@ -76,11 +77,19 @@ impl BroadcastedDeclareTransactionV2 {
         ])
         .into();
 
-        Ok(DeclareTransaction::new(
-            sn_api_declare,
-            starknet_api::transaction::TransactionHash(txn_hash.into()),
-            ContractClass::Cairo1(self.contract_class.clone()).try_into()?,
-        )?)
+        if FieldElement::from(self.common.version) > QUERY_VERSION_OFFSET {
+            Ok(DeclareTransaction::new_for_query(
+                sn_api_declare,
+                starknet_api::transaction::TransactionHash(txn_hash.into()),
+                ContractClass::Cairo1(self.contract_class.clone()).try_into()?,
+            )?)
+        } else {
+            Ok(DeclareTransaction::new(
+                sn_api_declare,
+                starknet_api::transaction::TransactionHash(txn_hash.into()),
+                ContractClass::Cairo1(self.contract_class.clone()).try_into()?,
+            )?)
+        }
```

Comment:
This if-else logic is repeated in at least three places. Could it be extracted to a common function?

Reply:
you mean the if condition -> self.common.version > QUERY_VERSION_OFFSET ?

Reply:
The common function I had in mind could accept `(version, tx, hash, class)` do the if-else logic based on version, and return `Result<DeclareTransaction, ...>`. Would that be an improvement?

Reply:
On which struct this function will be implemented

Reply:
I thought of only extracting the comparison in a function to keep the logic in 1 place

Reply:
yes, I had a regular function (non-method) in mind

Reply:
refactored